### PR TITLE
executor, plan: remove `Exists`

### PR DESCRIPTION
**Pull request by winoros**
## What have you changed? (mandatory)

We don't need this plan in fact.

## What is the type of the changes? (mandatory)

- improvement
- refactor

## How has this PR been tested? (mandatory)

existing tests.

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros please fix ci.

File: plan/expression_rewriter.go [hunk 1]
```diff
@@ -562,13 +562,36 @@ func (er *expressionRewriter) handleExistSubquery(v *ast.ExistsSubqueryExpr) (as
...
+		if len(rows) > 0 {
+			er.ctxStack = append(er.ctxStack, expression.One.Clone())
+		} else {
+			er.ctxStack = append(er.ctxStack, expression.Zero.Clone())
+		}
 	}
 	return v, true
 }
 
+func (er *expressionRewriter) popExistsSubPlan(p LogicalPlan) LogicalPlan {
```
Comment (XuHuaiyu) [inline plan/expression_rewriter.go]: add a comment for this function.

**Comment by zz-jason** [comment]
@winoros please resolve conflicts and add some labels for this PR.

**Comment by zhexuany** [comment]
@winoros there is a conflict, could you resolve it? Otherwise we cannot merge this PR into master.

**Pull request merged**